src/ol_infrastructure/infrastructure/aws/eks/__main__.py

```diff
@@ -104,38 +152,543 @@
+    )
+    aws.iam.RolePolicyAttachment(
+        f"{cluster_name}-efs-csi-driver-EFSCSIDriverPolicy-attachment",
+        policy_arn="arn:aws:iam::aws:policy/service-role/AmazonEFSCSIDriverPolicy",
+        role=efs_csi_driver_role.id,
+        opts=ResourceOptions(parent=efs_csi_driver_role),
+    )
+
+    efs_filesystem = aws.efs.FileSystem(
+        f"{cluster_name}-eks-filesystem",
+        encrypted=True,
+        kms_key_id=kms_ebs["arn"],
+        tags=aws_config.tags,
+        throughput_mode="bursting",
+        opts=ResourceOptions(parent=cluster, depends_on=cluster),
+    )
+
+    efs_security_group = aws.ec2.SecurityGroup(
+        f"{cluster_name}-eks-efs-securitygroup",
+        description="Allows the EKS subnets to access EFS",
+        vpc_id=target_vpc["id"],
+        ingress=[
+            aws.ec2.SecurityGroupIngressArgs(
+                protocol="tcp",
+                from_port=DEFAULT_EFS_PORT,
+                to_port=DEFAULT_EFS_PORT,
+                cidr_blocks=target_vpc["k8s_pod_subnet_cidrs"],
+                description=f"Allow traffic from EKS nodes on port {DEFAULT_EFS_PORT}",
+            ),
+        ],
+    )
+
+    def create_mountpoints(pod_subnet_ids):
+        for index, subnet_id in enumerate(pod_subnet_ids):
+            aws.efs.MountTarget(
+                f"{cluster_name}-eks-mounttarget-{index}",
+                file_system_id=efs_filesystem.id,
+                subnet_id=subnet_id,
+                security_groups=[efs_security_group.id],
+                opts=ResourceOptions(parent=efs_filesystem),
+            )
+
+    pod_subnet_ids.apply(lambda pod_subnet_ids: create_mountpoints(pod_subnet_ids))
+
+    kubernetes.storage.v1.StorageClass(
+        resource_name=f"{cluster_name}-efs-storageclass",
+        metadata=kubernetes.meta.v1.ObjectMetaArgs(
+            name="efs-sc",
+            labels=k8s_global_labels,
+        ),
+        provisioner="efs.csi.aws.com",
+        volume_binding_mode="WaitForFirstConsumer",
+        # ref: https://github.com/kubernetes-sigs/aws-efs-csi-driver/blob/master/docs/README.md
+        parameters={
+            "provisioningMode": "efs-ap",
+            "fileSystemId": efs_filesystem.id,
+            "directoryPerms": "700",
+        },
+        opts=ResourceOptions(
+            provider=k8s_provider,
+            parent=k8s_provider,
+            depends_on=[k8s_provider, efs_csi_driver_role],
+        ),
+    )
+    aws_efs_cni_driver_addon = eks.Addon(
+        f"{cluster_name}-eks-addon-efs-cni-driver-addon",
+        cluster=cluster,
+        addon_name="aws-efs-csi-driver",
+        addon_version=VERSIONS["EFS_CSI_DRIVER"],
+        service_account_role_arn=efs_csi_driver_role.arn,
+        opts=ResourceOptions(
+            parent=cluster,
+            # Addons won't install properly if there are not nodes to schedule them on
+            depends_on=[cluster, node_groups[0]],
+        ),
+    )
+
+# Configure and install any extra addons
```

Comment:
What does this mean? Would extra eks addons be addons configured elsewhere, like static configuration files or such? Just wondering about the use case.

Reply:
I think I originally imagined allowing us to define any "extra" addons beyond the storage providers as config yaml and then just creating the resources on fly using this loop. Turns out that we probably don't need to do that and there are not really any more interesting addons at the moment beyond the two we're doing explicitly for storage. So I will get rid of this.